pkg/cmd/cli/cmd/rsync/rsyncargs_test.go

```diff
@@ -0,0 +1,112 @@
+package rsync
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestRsyncArgsSet(t *testing.T) {
+	tests := []struct {
+		input       string
+		expected    []string
+		errExpected bool
+	}{
+		{
+			input:    "--progress --exclude='.git'",
+			expected: []string{"--progress", "--exclude=.git"},
+		},
+		{
+			input:    `--exclude='\*test.go' --include='integration'`,
+			expected: []string{"--exclude=\\*test.go", "--include=integration"},
+		},
+		{
+			input:       "--unknown=true",
+			errExpected: true,
+		},
+		{
+			input:       "--progress --unknown-flag",
+			errExpected: true,
+		},
+	}
+
+	for _, tc := range tests {
+		args := &RsyncArgs{}
+		err := args.Set(tc.input)
+		if err != nil && !tc.errExpected {
+			t.Errorf("Unexpected error: %v with input: %s", err, tc.input)
+			continue
+		}
+		if err == nil && tc.errExpected {
+			t.Errorf("Did not get expected error for input: %s", tc.input)
+			continue
+		}
+		if !reflect.DeepEqual([]string(*args), tc.expected) {
+			t.Errorf("Unexpected result: %#v for input: %s", args, tc.input)
+		}
+
+	}
+
+}
+
+func TestParseArgs(t *testing.T) {
+	tests := []struct {
+		input       string
+		expected    []string
+		errExpected bool
+	}{
+		{
+			input:    "hello world",
+			expected: []string{"hello", "world"},
+		},
+		{
+			input:    "\"hello world\"",
+			expected: []string{"hello world"},
+		},
+		{
+			input:    "one \"hello world\" two 'three'",
+			expected: []string{"one", "hello world", "two", "three"},
+		},
+		{
+			input:    "\"escaped \\\" quotes\" \"two\"",
+			expected: []string{"escaped \" quotes", "two"},
+		},
+		{
+			input:    "'single quote \\\" with escape' two",
+			expected: []string{"single quote \\\" with escape", "two"},
+		},
+		{
+			input:    "'multiple escaping'\"'\"' in same arg'",
+			expected: []string{"multiple escaping' in same arg"},
+		},
+		{
+			input:    "--test-arg=\"my value\" --another-arg='separate value'",
+			expected: []string{"--test-arg=my value", "--another-arg=separate value"},
```

Comment:
actually it is... because that's what the shell would pass  you. Remember, we're passing these as separate strings in the string array that's passed to the exec.Cmd. (they're no longer interpreted as multiple args)